### ISSUE TITLE
False positive on SC2016

### ISSUE THREAD

**Issue by asm0dey** [koalaman/shellcheck#1693]
#### For bugs
- Rule Id (if any, e.g. SC2016): 
- 0.7.0 and online: 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2016)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

set -x

YAML='name: "Portfolio optimization"
queue: default

master:
  resources:
    memory: 32 GiB
    vcores: '"$LMD_PO_NUM_CORES"'
  env:
    LMD_PO_OUTPUT_HDFS_PATH: '"$LMD_PO_OUTPUT_HDFS_PATH"'
    LMD_PO_INPUT_HDFS_PATH: '"$LMD_PO_INPUT_HDFS_PATH"'
    LMD_PO_NUM_CORES: '"$LMD_PO_NUM_CORES"'
  files:
    # The environment archive will be uploaded to HDFS, and extracted
    # into a directory named ``environment`` in each container.
    environment: environment.tar.gz
  script: |
    # Activate the environment
    source environment/bin/activate

    cd environment
    ./run.sh
'
TEMPFILE=$(mktemp)
echo "$YAML">"$TEMPFILE"

#snip

```

#### Here's what shellcheck currently says:

```
In skein-submit.sh line 15:
    LMD_PO_NUM_CORES: '"$LMD_PO_NUM_CORES"'
                                          ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
```

#### Here's what I wanted or expected to see:

No output for this line

**Comment by jwmrt** [comment]
See this related issue https://github.com/koalaman/shellcheck/issues/2464

**Comment by jwmrt** [comment]
The error is actually caused by the backticks on line 18 (as noted in this related issue https://github.com/koalaman/shellcheck/issues/2464 )
